lib/Dialect/TritonGPU/Transforms/Pipeliner/PipelineExpander.cpp

```diff
@@ -0,0 +1,683 @@
+//===- LoopPipelining.cpp - Code to perform loop software pipelining-------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements loop software pipelining
+//
+//===----------------------------------------------------------------------===//
+
+// Fork of upstream pipeliner. This will be merged upstream once things are
```

Comment:
I'm not quite clear about the exact code changes comparing this file with the upstream pipeline pass. Can you pin me when you prepare to commit changes to LLVM/MLIR?

Reply:
Definitely, I'll include you on the PRs.